packages/mask/src/site-adaptors/mirror.xyz/injection/Tips/TipsButtonWrapper.tsx

```diff
@@ -0,0 +1,70 @@
+import {
+    createInjectHooksRenderer,
+    Plugin,
+    useActivatedPluginsSiteAdaptor,
+    useIsMinimalMode,
+} from '@masknet/plugin-infra/content-script'
+import { EMPTY_LIST, PluginID, type SocialAccount } from '@masknet/shared-base'
+import { makeStyles } from '@masknet/theme'
+import type { Web3Helper } from '@masknet/web3-helpers'
+import { useNetworkContext, useWeb3Others } from '@masknet/web3-hooks-base'
+import { memo, useMemo } from 'react'
+import { useCurrentVisitingIdentity } from '../../../../components/DataSource/useActivatedUI.js'
+
+export const useStyles = makeStyles()((theme) => ({
+    root: {
+        position: 'relative',
+        marginLeft: theme.spacing(1),
+        height: 40,
+        width: 40,
+        display: 'flex',
+        justifyContent: 'center',
+        alignItems: 'center',
+        borderWidth: 1,
+        borderStyle: 'solid',
+        borderColor: theme.palette.maskColor.line,
+        borderRadius: 20,
+        marginRight: theme.spacing(1),
+        color: theme.palette.text.primary,
+    },
+}))
+
+interface Props {
+    slot: Plugin.SiteAdaptor.TipsSlot
+}
+export const TipsButtonWrapper = memo(function TipsButtonWrapper({ slot }: Props) {
+    const { classes } = useStyles()
+
+    const visitingIdentity = useCurrentVisitingIdentity()
+    const isMinimal = useIsMinimalMode(PluginID.Tips)
+    const { pluginID } = useNetworkContext()
+    const Others = useWeb3Others()
+
+    const accounts = useMemo((): Array<SocialAccount<Web3Helper.ChainIdAll>> => {
+        if (!visitingIdentity?.identifier) return EMPTY_LIST
+        return [
+            {
+                pluginID,
+                address: visitingIdentity.identifier.userId,
+                label: visitingIdentity.nickname
+                    ? `(${visitingIdentity.nickname}) ${Others.formatAddress(visitingIdentity.identifier.userId, 4)}`
+                    : visitingIdentity.identifier.userId,
+            },
+        ]
+    }, [visitingIdentity, Others.formatAddress])
+
+    const component = useMemo(() => {
+        const Component = createInjectHooksRenderer(
+            useActivatedPluginsSiteAdaptor.visibility.useNotMinimalMode,
+            (plugin) => plugin.TipsRealm?.UI?.Content,
+        )
+
+        return <Component identity={visitingIdentity.identifier} slot={slot} accounts={accounts} />
+    }, [visitingIdentity.identifier, accounts, slot])
+    const isDashboard = location.href === '/'
+    if (!component || !visitingIdentity.identifier || isMinimal || isDashboard) return null
+
+    return (
```

Comment:
It could yield span wraps block elements HTML.

Reply:
It's only the tip button, and elements inside are isolated.

BTW `span > div` is valid.